tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Collections/Array4Module.fs

```diff
@@ -23,12 +24,35 @@ Make sure each method works on:
 type Array4Module() =
 
     let VerifyDimensions arr x y z u =
-        if Array4D.length1 arr <> x then Assert.Fail("Array3D does not have expected dimensions.")
-        if Array4D.length2 arr <> y then Assert.Fail("Array3D does not have expected dimensions.")
-        if Array4D.length3 arr <> z then Assert.Fail("Array3D does not have expected dimensions.")
+        if Array4D.length1 arr <> x then Assert.Fail("Array4D does not have expected dimensions.")
+        if Array4D.length2 arr <> y then Assert.Fail("Array4D does not have expected dimensions.")
+        if Array4D.length3 arr <> z then Assert.Fail("Array4D does not have expected dimensions.")
```

Comment:
Rather than failing with an uninformative error, can you provide output that will show what was expected versus what was observed?

Reply:
Yep, that's a good idea